### PR TITLE
Make deployment plans unique

### PR DESCRIPTION
**Pull request by deanmarcussen**
Fixes https://github.com/OrchardCMS/OrchardCore/issues/6774

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs [hunk 1]
```diff
@@ -277,6 +283,14 @@ public async Task<IActionResult> Edit(EditDeploymentPlanViewModel model)
                 {
                     ModelState.AddModelError(nameof(EditDeploymentPlanViewModel.Name), S["The name is mandatory."]);
                 }
+                if (model.Name != deploymentPlan.Name)
+                {
+                    var count = await _session.QueryIndex<DeploymentPlanIndex>(x => x.Name == model.Name && x.DocumentId != model.Id).CountAsync();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs:288]: Please use `.Equals()` with ignore casing

**Review by hishamco — CHANGES_REQUESTED** [review]
LGTM after we ignore the casing

File: src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs (hunk 1 above)
Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs:288]: Yeah makes sense. Done

**Comment by agriffard** [comment]
Build error: 
`DeploymentPlanController.cs(286,69): error CS1503: Argument 3: cannot convert from 'System.StringComparer' to 'System.StringComparison'`

**Comment by agriffard** [comment]
Test run is still failing after the change.

**Pull request merged**